<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2025-05-23" />

<title>E18 mLN pLN Ccl19EYFP+</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">LNdevMouse24.2</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">E18 mLN pLN Ccl19EYFP+</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2025-05-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>LNdevMouse24.2/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250625code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250625)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250625code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250625)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLudewigLabLNdevMouse242treef7184edad716c9d4cb8f03efec5153b9ab4009eetargetblankf7184eda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Ludewig-Lab/LNdevMouse24.2/tree/f7184edad716c9d4cb8f03efec5153b9ab4009ee" target="_blank">f7184ed</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLudewigLabLNdevMouse242treef7184edad716c9d4cb8f03efec5153b9ab4009eetargetblankf7184eda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Ludewig-Lab/LNdevMouse24.2/tree/f7184edad716c9d4cb8f03efec5153b9ab4009ee" target="_blank">f7184ed</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    data/to figshare/
    Ignored:    data/tradeSEQ/

Untracked files:
    Untracked:  analysis/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd

Unstaged changes:
    Modified:   analysis/E18_mLN_iLN_EYFPonly_marker.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="preprocess" class="section level3">
<h3>preprocess</h3>
<div id="load-packages" class="section level4">
<h4>load packages</h4>
<pre class="r"><code>library(ExploreSCdataSeurat3)
library(runSeurat3)
library(Seurat)
library(ggpubr)
library(pheatmap)
library(SingleCellExperiment)
library(dplyr)
library(tidyverse)
library(viridis)
library(muscat)
library(circlize)
library(destiny)
library(scater)
library(metap)
library(multtest)
library(clusterProfiler)
library(org.Mm.eg.db)
library(msigdbr)
library(enrichplot)
library(DOSE)
library(grid)
library(gridExtra)
library(ggupset)
library(VennDiagram)
library(NCmisc)
library(slingshot)
library(here)
library(RColorBrewer)</code></pre>
</div>
<div id="set-color-vectors" class="section level4">
<h4>set color vectors</h4>
<pre class="r"><code>coltimepoint &lt;- c(&quot;#440154FF&quot;, &quot;#3B528BFF&quot;, &quot;#21908CFF&quot;, &quot;#5DC863FF&quot;)
names(coltimepoint) &lt;- c(&quot;E18&quot;, &quot;P7&quot;, &quot;3w&quot;, &quot;8w&quot;)

collocation &lt;- c(&quot;#61baba&quot;, &quot;#ba6161&quot;)
names(collocation) &lt;- c(&quot;iLN&quot;, &quot;mLN&quot;)</code></pre>
</div>
</div>
<div id="load-object-all" class="section level3">
<h3>load object all</h3>
<pre class="r"><code>basedir &lt;- here()
fileNam &lt;- paste0(basedir, &quot;/data/AllSamplesMerged_seurat.rds&quot;)
seuratM &lt;- readRDS(fileNam)
table(seuratM$age)</code></pre>
<pre><code>
      3w       8w E17to7wk      E18       P7 
   12291    32503    19685    33903    10005 </code></pre>
<div id="subset-e18" class="section level4">
<h4>subset E18</h4>
<pre class="r"><code>seuratA &lt;- subset(seuratM, age == &quot;E18&quot;)
table(seuratA$age)</code></pre>
<pre><code>
  E18 
33903 </code></pre>
<pre class="r"><code>## rerun seurat
seuratA &lt;- NormalizeData (object = seuratA)
seuratA &lt;- FindVariableFeatures(object = seuratA)
seuratA &lt;- ScaleData(object = seuratA, verbose = TRUE)
seuratA &lt;- RunPCA(object=seuratA, npcs = 30, verbose = FALSE)
seuratA &lt;- RunTSNE(object=seuratA, reduction=&quot;pca&quot;, dims = 1:20)
seuratA &lt;- RunUMAP(object=seuratA, reduction=&quot;pca&quot;, dims = 1:20)
seuratA &lt;- FindNeighbors(object = seuratA, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratA &lt;- FindClusters(object = seuratA, resolution = res[i], random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33903
Number of edges: 1106453

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9482
Number of communities: 16
Elapsed time: 6 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33903
Number of edges: 1106453

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9122
Number of communities: 20
Elapsed time: 6 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33903
Number of edges: 1106453

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8973
Number of communities: 25
Elapsed time: 5 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33903
Number of edges: 1106453

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9310
Number of communities: 18
Elapsed time: 6 seconds</code></pre>
</div>
</div>
<div id="dimplot-all" class="section level3">
<h3>dimplot all</h3>
<div id="clustering" class="section level4">
<h4>clustering</h4>
<pre class="r"><code>Idents(seuratA) &lt;- seuratA$RNA_snn_res.0.25
DimPlot(seuratA, reduction = &quot;umap&quot;, group.by = &quot;RNA_snn_res.0.25&quot; ,
        pt.size = 0.1, label = T, shuffle = T) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20clustering%20pre%20filter-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="location" class="section level4">
<h4>location</h4>
<pre class="r"><code>DimPlot(seuratA, reduction = &quot;umap&quot;, group.by = &quot;location&quot; ,
        pt.size = 0.1, label = T, shuffle = T) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20location%20pre%20filter-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="features" class="section level4">
<h4>features</h4>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratA)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

selGenes &lt;- data.frame(geneID=c(&quot;Ptprc&quot;, &quot;Msln&quot;, &quot;Mki67&quot;, &quot;Kcnn3&quot;, &quot;Tcf21&quot;, &quot;Pecam1&quot;, &quot;Lyve1&quot;, &quot;Ccl21a&quot;, &quot;Icam1&quot;, &quot;Cd34&quot;, &quot;Mrc1&quot;, &quot;Rosa26eyfp&quot;, &quot;Krt19&quot;, &quot;Acta2&quot;,&quot;Myh11&quot;, &quot;Cxcl13&quot;, &quot;Epcam&quot;, &quot;Adipoq&quot;)) %&gt;% 
  left_join(., genes, by = &quot;geneID&quot;) 

pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- FeaturePlot(seuratA, reduction = &quot;umap&quot;, 
            features = x, 
            cols=c(&quot;lightgrey&quot;,&quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-16.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20sel%20genes%20pre%20filter-18.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="filter" class="section level4">
<h4>filter</h4>
<pre class="r"><code>## filter and Pecam+ cells (cluster 15)
## filter immune cells (cluseter 14)
table(seuratA$RNA_snn_res.0.25)
seuratF &lt;- subset(seuratA, RNA_snn_res.0.25 %in% c(&quot;15&quot;, &quot;14&quot;), invert = TRUE)
table(seuratF$RNA_snn_res.0.25)

seuratE18fil &lt;- seuratF
remove(seuratF)</code></pre>
</div>
<div id="rerun-after-fil" class="section level4">
<h4>rerun after fil</h4>
</div>
<div id="load-object-fil" class="section level4">
<h4>load object fil</h4>
<pre class="r"><code>fileNam &lt;- paste0(basedir, &quot;/data/E18fil_seurat.rds&quot;)
seuratE18fil &lt;- readRDS(fileNam)</code></pre>
</div>
</div>
<div id="dimplot-e18-fil" class="section level3">
<h3>dimplot E18 fil</h3>
<div id="clustering-1" class="section level4">
<h4>clustering</h4>
<pre class="r"><code>Idents(seuratE18fil) &lt;- seuratE18fil$RNA_snn_res.0.25
colPal &lt;- c(&quot;#DAF7A6&quot;, &quot;#FFC300&quot;, &quot;#FF5733&quot;, &quot;#C70039&quot;, &quot;#900C3F&quot;, &quot;#b66e8d&quot;,
            &quot;#61a4ba&quot;, &quot;#6178ba&quot;, &quot;#54a87f&quot;,  &quot;#25328a&quot;,
            &quot;#b6856e&quot;, &quot;#0073C2FF&quot;, &quot;#EFC000FF&quot;, &quot;#868686FF&quot;, &quot;#CD534CFF&quot;, 
            &quot;#7AA6DCFF&quot;, &quot;#003C67FF&quot;, &quot;#8F7700FF&quot;, &quot;#3B3B3BFF&quot;, &quot;#A73030FF&quot;,
            &quot;#4A6990FF&quot;)[1:length(unique(seuratE18fil$RNA_snn_res.0.25))]
names(colPal) &lt;- unique(seuratE18fil$RNA_snn_res.0.25)

DimPlot(seuratE18fil, reduction = &quot;umap&quot;, group.by = &quot;RNA_snn_res.0.25&quot;,
        cols = colPal, label = TRUE)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/dimplot%20clustering%20E18fil-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="location-1" class="section level4">
<h4>location</h4>
<pre class="r"><code>DimPlot(seuratE18fil, reduction = &quot;umap&quot;, group.by = &quot;location&quot;,
        cols = collocation)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/dimplot%20loction%20E18fil-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(seuratE18fil, file=paste0(basedir, &quot;/data/E18fil_seurat.rds&quot;))</code></pre>
</div>
</div>
<div id="subset-eyfp-expressing-cells" class="section level3">
<h3>subset EYFP expressing cells</h3>
<pre class="r"><code>table(seuratE18fil$EYFP)
seuratE18EYFPv2 &lt;- subset(seuratE18fil, EYFP == &quot;pos&quot;)
table(seuratE18EYFPv2$EYFP)

DimPlot(seuratE18EYFPv2, reduction = &quot;umap&quot;, group.by = &quot;RNA_snn_res.0.25&quot;,
        cols = colPal, label = TRUE)

#rerun seurat
seuratE18EYFPv2 &lt;- NormalizeData (object = seuratE18EYFPv2)
seuratE18EYFPv2&lt;- FindVariableFeatures(object = seuratE18EYFPv2)
seuratE18EYFPv2 &lt;- ScaleData(object = seuratE18EYFPv2, verbose = TRUE)
seuratE18EYFPv2 &lt;- RunPCA(object=seuratE18EYFPv2, npcs = 30, verbose = FALSE)
seuratE18EYFPv2 &lt;- RunTSNE(object=seuratE18EYFPv2, reduction=&quot;pca&quot;, dims = 1:20)
seuratE18EYFPv2 &lt;- RunUMAP(object=seuratE18EYFPv2, reduction=&quot;pca&quot;, dims = 1:20)
seuratE18EYFPv2 &lt;- FindNeighbors(object = seuratE18EYFPv2, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratE18EYFPv2 &lt;- FindClusters(object = seuratE18EYFPv2, resolution = res[i], random.seed = 1234)
}</code></pre>
<pre class="r"><code>saveRDS(seuratE18EYFPv2, file=paste0(basedir, &quot;/data/E18_EYFPv2_Ccl19_seurat.rds&quot;))</code></pre>
</div>
<div id="load-object-e18-eyfp" class="section level3">
<h3>load object E18 EYFP+</h3>
<pre class="r"><code>fileNam &lt;- paste0(basedir, &quot;/data/E18_EYFPv2_Ccl19_seurat.rds&quot;)
seuratE18EYFPv2 &lt;- readRDS(fileNam)</code></pre>
</div>
<div id="dimplot-e18-eyfp" class="section level3">
<h3>dimplot E18 EYFP+</h3>
<div id="clustering-2" class="section level4">
<h4>clustering</h4>
<pre class="r"><code>Idents(seuratE18EYFPv2) &lt;- seuratE18EYFPv2$RNA_snn_res.0.25
colPal &lt;- c(&quot;#DAF7A6&quot;, &quot;#FFC300&quot;, &quot;#FF5733&quot;, &quot;#C70039&quot;, &quot;#900C3F&quot;, &quot;#b66e8d&quot;,
            &quot;#61a4ba&quot;, &quot;#6178ba&quot;, &quot;#54a87f&quot;,  &quot;#25328a&quot;,
            &quot;#b6856e&quot;, &quot;#0073C2FF&quot;, &quot;#EFC000FF&quot;, &quot;#868686FF&quot;, &quot;#CD534CFF&quot;, 
            &quot;#7AA6DCFF&quot;, &quot;#003C67FF&quot;, &quot;#8F7700FF&quot;, &quot;#3B3B3BFF&quot;, &quot;#A73030FF&quot;,
            &quot;#4A6990FF&quot;)[1:length(unique(seuratE18EYFPv2$RNA_snn_res.0.25))]
names(colPal) &lt;- unique(seuratE18EYFPv2$RNA_snn_res.0.25)

DimPlot(seuratE18EYFPv2, reduction = &quot;umap&quot;, group.by = &quot;RNA_snn_res.0.25&quot;,
        cols = colPal, label = TRUE)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/dimplot%20clustering%20E18%20EYFP%20pos%20v2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="location-2" class="section level4">
<h4>location</h4>
<pre class="r"><code>DimPlot(seuratE18EYFPv2, reduction = &quot;umap&quot;, group.by = &quot;location&quot;,
        cols = collocation)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/dimplot%20loction%20E18%20EYFP%20v2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="features-e18-eyfp" class="section level3">
<h3>features E18 EYFP+</h3>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratE18EYFPv2)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

selGenes &lt;- data.frame(geneID=c(&quot;Rosa26eyfp&quot;, &quot;Mki67&quot;, &quot;Acta2&quot;, &quot;Myh11&quot;, &quot;Ccl19&quot;, &quot;Cxcl13&quot;, &quot;Cd34&quot;, &quot;Icam1&quot;,&quot;Vcam1&quot;, &quot;Pi16&quot;)) %&gt;% 
  left_join(., genes, by = &quot;geneID&quot;) 

pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- FeaturePlot(seuratE18EYFPv2, reduction = &quot;umap&quot;, 
            features = x, 
            cols=c(&quot;lightgrey&quot;,&quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/featureplot%20E18%20EYFP%20only%20v2-10.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="integrate-data-across-location" class="section level3">
<h3>integrate data across location</h3>
<pre class="r"><code>Idents(seuratE18EYFPv2) &lt;- seuratE18EYFPv2$location

seurat.list &lt;- SplitObject(object = seuratE18EYFPv2, split.by = &quot;location&quot;)
for (i in 1:length(x = seurat.list)) {
    seurat.list[[i]] &lt;- NormalizeData(object = seurat.list[[i]],
                                      verbose = FALSE)
    seurat.list[[i]] &lt;- FindVariableFeatures(object = seurat.list[[i]], 
        selection.method = &quot;vst&quot;, nfeatures = 2000, verbose = FALSE)
}

seurat.anchors &lt;- FindIntegrationAnchors(object.list = seurat.list, dims = 1:20)
seuratE18EYFPv2.int &lt;- IntegrateData(anchorset = seurat.anchors, dims = 1:20)
DefaultAssay(object = seuratE18EYFPv2.int) &lt;- &quot;integrated&quot;

## rerun seurat
seuratE18EYFPv2.int &lt;- ScaleData(object = seuratE18EYFPv2.int, verbose = FALSE,
                         features = rownames(seuratE18EYFPv2.int))
seuratE18EYFPv2.int &lt;- RunPCA(object = seuratE18EYFPv2.int, npcs = 20, verbose = FALSE)
seuratE18EYFPv2.int &lt;- RunTSNE(object = seuratE18EYFPv2.int, recuction = &quot;pca&quot;, dims = 1:20)
seuratE18EYFPv2.int &lt;- RunUMAP(object = seuratE18EYFPv2.int, recuction = &quot;pca&quot;, dims = 1:20)

seuratE18EYFPv2.int &lt;- FindNeighbors(object = seuratE18EYFPv2.int, reduction = &quot;pca&quot;, dims = 1:20)
res &lt;- c(0.6, 0.8, 0.4, 0.25)
for (i in 1:length(res)){
  seuratE18EYFPv2.int &lt;- FindClusters(object = seuratE18EYFPv2.int, resolution = res[i],
                              random.seed = 1234)
}</code></pre>
<pre class="r"><code>saveRDS(seuratE18EYFPv2.int, file=paste0(basedir, &quot;/data/E18EYFPv2_integrated_Ccl19_seurat.rds&quot;))</code></pre>
</div>
<div id="load-object-e18-eyfp-integrated" class="section level3">
<h3>load object E18 EYFP+ integrated</h3>
<pre class="r"><code>fileNam &lt;- paste0(basedir, &quot;/data/E18EYFPv2_integrated_Ccl19_seurat.rds&quot;)
seuratE18EYFPv2.int &lt;- readRDS(fileNam)</code></pre>
<pre class="r"><code>DefaultAssay(object = seuratE18EYFPv2.int) &lt;- &quot;RNA&quot;
seuratE18EYFPv2.int$intCluster &lt;- seuratE18EYFPv2.int$integrated_snn_res.0.25
Idents(seuratE18EYFPv2.int) &lt;- seuratE18EYFPv2.int$intCluster

colPal &lt;- c(&quot;#DAF7A6&quot;, &quot;#FFC300&quot;, &quot;#FF5733&quot;, &quot;#C70039&quot;, &quot;#900C3F&quot;, &quot;#b66e8d&quot;,
            &quot;#61a4ba&quot;, &quot;#6178ba&quot;, &quot;#54a87f&quot;, &quot;#25328a&quot;, &quot;#b6856e&quot;,
            &quot;#ba6161&quot;, &quot;#20714a&quot;, &quot;#0073C2FF&quot;, &quot;#EFC000FF&quot;, &quot;#868686FF&quot;, 
            &quot;#CD534CFF&quot;,&quot;#7AA6DCFF&quot;, &quot;#003C67FF&quot;, &quot;#8F7700FF&quot;, &quot;#3B3B3BFF&quot;,
            &quot;#A73030FF&quot;, &quot;#4A6990FF&quot;)[1:length(unique(seuratE18EYFPv2.int$intCluster))]
names(colPal) &lt;- unique(seuratE18EYFPv2.int$intCluster)</code></pre>
</div>
<div id="dimplot-e18-eyfp-int" class="section level3">
<h3>dimplot E18 EYFP+ int</h3>
<div id="clustering-3" class="section level4">
<h4>clustering</h4>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;,
        label = T, shuffle = T, cols = colPal) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20v2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="location-3" class="section level4">
<h4>location</h4>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, group.by = &quot;location&quot;, cols = collocation,
        shuffle = T) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20location%20v2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="assign-label" class="section level4">
<h4>assign label</h4>
<pre class="r"><code>seuratE18EYFPv2.int$label &lt;- &quot;label&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;0&quot;)] &lt;- &quot;cluster1&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;1&quot;)] &lt;- &quot;Prolif&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;2&quot;)] &lt;- &quot;cluster3&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;3&quot;)] &lt;- &quot;cluster4&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;4&quot;)] &lt;- &quot;cluster2&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;5&quot;)] &lt;- &quot;Adipoc&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;6&quot;)] &lt;- &quot;Mesothelial&quot;

table(seuratE18EYFPv2.int$label)</code></pre>
<pre><code>
     Adipoc    cluster1    cluster2    cluster3    cluster4 Mesothelial      Prolif 
         17         852         292         514         450          15         591 </code></pre>
<pre class="r"><code>##order
seuratE18EYFPv2.int$label &lt;- factor(seuratE18EYFPv2.int$label, levels = c(&quot;cluster1&quot;, &quot;cluster2&quot;, &quot;cluster3&quot;, &quot;cluster4&quot;, &quot;Adipoc&quot;, &quot;Mesothelial&quot;, &quot;Prolif&quot;))
table(seuratE18EYFPv2.int$label)</code></pre>
<pre><code>
   cluster1    cluster2    cluster3    cluster4      Adipoc Mesothelial      Prolif 
        852         292         514         450          17          15         591 </code></pre>
<pre class="r"><code>colLab &lt;- c(&quot;#900C3F&quot;,&quot;#b66e8d&quot;, &quot;#003C67FF&quot;,
            &quot;#e3953d&quot;, &quot;#714542&quot;, &quot;#b6856e&quot;,&quot;grey&quot;)
names(colLab) &lt;- c(&quot;cluster1&quot;, &quot;cluster2&quot;, &quot;cluster3&quot;, &quot;cluster4&quot;, &quot;Adipoc&quot;, &quot;Mesothelial&quot;, &quot;Prolif&quot;)</code></pre>
</div>
<div id="label" class="section level4">
<h4>label</h4>
<pre class="r"><code>colLab &lt;- c(&quot;#900C3F&quot;,&quot;#b66e8d&quot;, &quot;#003C67FF&quot;,
            &quot;#e3953d&quot;, &quot;#714542&quot;, &quot;#b6856e&quot;,&quot;grey&quot;)
names(colLab) &lt;- c(&quot;cluster1&quot;, &quot;cluster2&quot;, &quot;cluster3&quot;, &quot;cluster4&quot;, &quot;Adipoc&quot;, &quot;Mesothelial&quot;, &quot;Prolif&quot;)

DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols = colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20label%20v2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, pt.size=0.5,
        cols = colLab, shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20label%20v2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, pt.size=0.5,
        cols = colLab, shuffle = T)+
  theme_void() + 
  theme(legend.position = &quot;none&quot;) </code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20label%20v2-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dotplot-frc-marker-e18-eyfp-int" class="section level3">
<h3>dotplot FRC marker E18 EYFP+ int</h3>
<pre class="r"><code>seurat_markers &lt;- data.frame(gene=c(&quot;Vcam1&quot;, &quot;Icam1&quot;,
                                    &quot;Cxcl13&quot;, &quot;Ccl19&quot;, &quot;Ccl21a&quot;,&quot;Tnfsf11&quot;, &quot;Grem1&quot;,&quot;Ifitm1&quot;,&quot;Cxcl1&quot;,&quot;Ifitm3&quot;,&quot;Ccl2&quot;,&quot;Nfkbia&quot;,&quot;Des&quot;,
                                    &quot;Mfap5&quot;,&quot;Cdkn1c&quot;,&quot;Akap12&quot;,&quot;Anxa2&quot;,&quot;Lox&quot;,&quot;Gsn&quot;,&quot;Basp1&quot;,&quot;Fndc1&quot;,&quot;Sparc&quot;,&quot;Col1a1&quot;,&quot;Fbn2&quot;,&quot;Nr4a1&quot;,&quot;Fbln1&quot;,&quot;Cd34&quot;,&quot;Pi16&quot;,
                                    &quot;Fbln5&quot;,&quot;Tm4sf1&quot;, &quot;Ntrk3&quot;, &quot;Fhl1&quot;, &quot;Rgs7bp&quot;, &quot;Adamts2&quot;, &quot;Mpped2&quot;, &quot;Ramp1&quot;, &quot;Pdgfrl&quot;, &quot;Eln&quot;, &quot;Hspb2&quot;,&quot;Mgp&quot;, &quot;Actg2&quot;,&quot;Acta2&quot;, &quot;Myh11&quot;,&quot;Mcam&quot;,&quot;Adipoq&quot;, &quot;Mki67&quot;, &quot;Msln&quot;))

genes &lt;- data.frame(geneID=rownames(seuratE18EYFPv2.int)) %&gt;%
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) 

markerAll &lt;- seurat_markers %&gt;% left_join(., genes, by=&quot;gene&quot;)


## Dotplot all
Idents(seuratE18EYFPv2.int) &lt;- seuratE18EYFPv2.int$label

DotPlot(seuratE18EYFPv2.int, assay=&quot;RNA&quot;, features = rev(markerAll$geneID), scale =T,
        cluster.idents = F) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=rev(markerAll$geneID), labels=rev(markerAll$gene)) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/dotplot%20FRC%20marker%20v2-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="subset-frcs-and-rerun" class="section level3">
<h3>subset FRCs and rerun</h3>
<pre class="r"><code>table(seuratE18EYFPv2.int$label)</code></pre>
<pre><code>
   cluster1    cluster2    cluster3    cluster4      Adipoc Mesothelial      Prolif 
        852         292         514         450          17          15         591 </code></pre>
<pre class="r"><code>seuratE18EYFPv2.int &lt;- subset(seuratE18EYFPv2.int, label %in% c(&quot;Adipoc&quot;, &quot;Mesothelial&quot;), invert = TRUE)
table(seuratE18EYFPv2.int$label)</code></pre>
<pre><code>
cluster1 cluster2 cluster3 cluster4   Prolif 
     852      292      514      450      591 </code></pre>
<pre class="r"><code>table(seuratE18EYFPv2.int$orig.ident)</code></pre>
<pre><code>
     
2699 </code></pre>
<pre class="r"><code>## rerun seurat
DefaultAssay(object = seuratE18EYFPv2.int) &lt;- &quot;integrated&quot;
seuratE18EYFPv2.int &lt;- ScaleData(object = seuratE18EYFPv2.int, verbose = FALSE,
                         features = rownames(seuratE18EYFPv2.int))
seuratE18EYFPv2.int &lt;- RunPCA(object = seuratE18EYFPv2.int, npcs = 20, verbose = FALSE)
seuratE18EYFPv2.int &lt;- RunTSNE(object = seuratE18EYFPv2.int, recuction = &quot;pca&quot;, dims = 1:20)
seuratE18EYFPv2.int &lt;- RunUMAP(object = seuratE18EYFPv2.int, recuction = &quot;pca&quot;, dims = 1:20)

seuratE18EYFPv2.int &lt;- FindNeighbors(object = seuratE18EYFPv2.int, reduction = &quot;pca&quot;, dims = 1:20)
res &lt;- c(0.1, 0.6, 0.8, 0.4, 0.25)
for (i in 1:length(res)){
  seuratE18EYFPv2.int &lt;- FindClusters(object = seuratE18EYFPv2.int, resolution = res[i],
                              random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2699
Number of edges: 93881

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9410
Number of communities: 3
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2699
Number of edges: 93881

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8333
Number of communities: 8
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2699
Number of edges: 93881

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8082
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2699
Number of edges: 93881

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8679
Number of communities: 8
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2699
Number of edges: 93881

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8977
Number of communities: 5
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>DefaultAssay(object = seuratE18EYFPv2.int) &lt;- &quot;RNA&quot;
seuratE18EYFPv2.int$intCluster &lt;- seuratE18EYFPv2.int$integrated_snn_res.0.1
Idents(seuratE18EYFPv2.int) &lt;- seuratE18EYFPv2.int$intCluster

colPal &lt;- c(&quot;#DAF7A6&quot;, &quot;#FFC300&quot;, &quot;#FF5733&quot;, &quot;#C70039&quot;, &quot;#900C3F&quot;, &quot;#b66e8d&quot;,
            &quot;#61a4ba&quot;, &quot;#6178ba&quot;, &quot;#54a87f&quot;, &quot;#25328a&quot;, &quot;#b6856e&quot;,
            &quot;#ba6161&quot;, &quot;#20714a&quot;, &quot;#0073C2FF&quot;, &quot;#EFC000FF&quot;, &quot;#868686FF&quot;, 
            &quot;#CD534CFF&quot;,&quot;#7AA6DCFF&quot;, &quot;#003C67FF&quot;, &quot;#8F7700FF&quot;, &quot;#3B3B3BFF&quot;,
            &quot;#A73030FF&quot;, &quot;#4A6990FF&quot;)[1:length(unique(seuratE18EYFPv2.int$intCluster))]
names(colPal) &lt;- unique(seuratE18EYFPv2.int$intCluster)</code></pre>
</div>
<div id="dimplot-e18-eyfp-fil" class="section level3">
<h3>dimplot E18 EYFP+ fil</h3>
<div id="clustering-4" class="section level4">
<h4>clustering</h4>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;,
        label = T, shuffle = T, cols = colPal) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20v2%20mLto%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="location-4" class="section level4">
<h4>location</h4>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, group.by = &quot;location&quot;, cols = collocation,
        shuffle = T) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20location%20v2%20mLTo%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="assign-label-1" class="section level4">
<h4>assign label</h4>
<pre class="r"><code>seuratE18EYFPv2.int$label &lt;- &quot;label&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;0&quot;)] &lt;- &quot;cluster2&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;1&quot;)] &lt;- &quot;cluster3&quot;
seuratE18EYFPv2.int$label[which(seuratE18EYFPv2.int$intCluster == &quot;2&quot;)] &lt;- &quot;cluster1&quot;

table(seuratE18EYFPv2.int$label)</code></pre>
<pre><code>
cluster1 cluster2 cluster3 
     760     1164      775 </code></pre>
<pre class="r"><code>##order
seuratE18EYFPv2.int$label &lt;- factor(seuratE18EYFPv2.int$label, levels = c(&quot;cluster2&quot;, &quot;cluster3&quot;, &quot;cluster1&quot;))
table(seuratE18EYFPv2.int$label)</code></pre>
<pre><code>
cluster2 cluster3 cluster1 
    1164      775      760 </code></pre>
<pre class="r"><code>colLab &lt;- c(&quot;#900C3F&quot;,&quot;#b66e8d&quot;, &quot;#003C67FF&quot;)
names(colLab) &lt;- c(&quot;cluster2&quot;, &quot;cluster3&quot;, &quot;cluster1&quot;)</code></pre>
</div>
<div id="label-1" class="section level4">
<h4>label</h4>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols = colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20label%20v2%20mLTo%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, pt.size=0.5,
        cols = colLab, shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20label%20v2%20mLTo%20only-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, pt.size=0.5,
        cols = colLab, shuffle = T)+
  theme_void() + 
  theme(legend.position = &quot;none&quot;) </code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/umap%20int%20label%20v2%20mLTo%20only-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="featrue-ccl19" class="section level3">
<h3>featrue Ccl19</h3>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratE18EYFPv2.int)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

selGenes &lt;- data.frame(geneID=c(&quot;Ccl19&quot;)) %&gt;% 
  left_join(., genes, by = &quot;geneID&quot;) 

pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- FeaturePlot(seuratE18EYFPv2.int, reduction = &quot;umap&quot;, 
            features = x, 
            cols=c(&quot;lightgrey&quot;,&quot;darkred&quot;),
            order = FALSE)+
  theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/feature%20Ccl19-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="dotplot-marker-e18-eyfp-fil" class="section level2">
<h2>dotplot marker E18 EYFP+ fil</h2>
<pre class="r"><code>seurat_markers &lt;- data.frame(gene=c(&quot;Vcam1&quot;, &quot;Icam1&quot;,
                                    &quot;Cxcl13&quot;, &quot;Ccl19&quot;, &quot;Ccl21a&quot;,&quot;Tnfsf11&quot;, &quot;Grem1&quot;,&quot;Ifitm1&quot;,&quot;Cxcl1&quot;,&quot;Ifitm3&quot;,&quot;Ccl2&quot;,&quot;Nfkbia&quot;,&quot;Des&quot;,
                                    &quot;Mfap5&quot;,&quot;Cdkn1c&quot;,&quot;Akap12&quot;,&quot;Anxa2&quot;,&quot;Lox&quot;,&quot;Gsn&quot;,&quot;Basp1&quot;,&quot;Fndc1&quot;,&quot;Sparc&quot;,&quot;Col1a1&quot;,&quot;Fbn2&quot;,&quot;Nr4a1&quot;,&quot;Fbln1&quot;,&quot;Cd34&quot;,&quot;Pi16&quot;,
                                    &quot;Fbln5&quot;,&quot;Tm4sf1&quot;, &quot;Ntrk3&quot;, &quot;Fhl1&quot;, &quot;Rgs7bp&quot;, &quot;Adamts2&quot;, &quot;Mpped2&quot;, &quot;Ramp1&quot;, &quot;Pdgfrl&quot;, &quot;Eln&quot;, &quot;Hspb2&quot;,&quot;Mgp&quot;, &quot;Actg2&quot;,&quot;Acta2&quot;, &quot;Myh11&quot;, &quot;Mcam&quot;, &quot;Mki67&quot;, &quot;Ccna2&quot;, &quot;Cdca8&quot;, &quot;Prc1&quot;, &quot;Aurkb&quot;))

genes &lt;- data.frame(geneID=rownames(seuratE18EYFPv2.int)) %&gt;%
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) 

markerAll &lt;- seurat_markers %&gt;% left_join(., genes, by=&quot;gene&quot;)


## Dotplot all
Idents(seuratE18EYFPv2.int) &lt;- seuratE18EYFPv2.int$label

DotPlot(seuratE18EYFPv2.int, assay=&quot;RNA&quot;, features = rev(markerAll$geneID), scale =T,
        cluster.idents = F) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=rev(markerAll$geneID), labels=rev(markerAll$gene)) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/dotplot%20FRC%20marker%20v2%20mLTo%20only-1.png" width="576" style="display: block; margin: auto;" /></p>
<div id="signatures" class="section level3">
<h3>signatures</h3>
<pre class="r"><code>## convert seurat object to sce object
## exteract logcounts
logcounts &lt;- GetAssayData(seuratE18EYFPv2.int, assay = &quot;RNA&quot;, slot = &quot;data&quot;)
counts &lt;- GetAssayData(seuratE18EYFPv2.int, assay = &quot;RNA&quot;, slot = &quot;counts&quot;)
## extract reduced dims from integrated assay
pca &lt;- Embeddings(seuratE18EYFPv2.int, reduction = &quot;pca&quot;)
umap &lt;- Embeddings(seuratE18EYFPv2.int, reduction = &quot;umap&quot;)
## create sce object
sce &lt;- SingleCellExperiment(assays =list (
  counts = counts,
  logcounts = logcounts
),
colData = seuratE18EYFPv2.int@meta.data,
rowData = data.frame(gene_id = rownames(logcounts)),
reducedDims = SimpleList(
  PCA = pca,
  UMAP = umap
))

genes &lt;- data.frame(geneID=rownames(sce)) %&gt;% mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;, geneID))
pal = colorRampPalette(c(&quot;#053061&quot;, &quot;#2166ac&quot;, &quot;#f7f7f7&quot;, &quot;#f4a582&quot;, &quot;#b2183c&quot;, &quot;#85122d&quot;))</code></pre>
<pre class="r"><code>selGenes &lt;- data.frame(gene=c(&quot;Cxcl13&quot;, &quot;Ccl19&quot;, &quot;Ccl21a&quot;,&quot;Tnfsf11&quot;, &quot;Grem1&quot;))
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% selGenes$gene)

##make a count matrix of signature genes
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 3))
sceSub$sign2[which(sceSub$sign &gt; 3)] &lt;- 3
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 2.890401</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/signature%20cluster2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/signature%20cluster2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>selGenes &lt;- data.frame(gene=c(&quot;Mfap5&quot;,&quot;Gsn&quot;,&quot;Fndc1&quot;,&quot;Col1a1&quot;,&quot;Cd34&quot;))
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% selGenes$gene)

##make a count matrix of signature genes
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 3))
sceSub$sign2[which(sceSub$sign &gt; 3)] &lt;- 3
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 3.508908</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/signature%20cluster3%20-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/signature%20cluster3%20-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>selGenes &lt;- data.frame(gene=c(&quot;Fbln5&quot;,&quot;Eln&quot;,&quot;Actg2&quot;,&quot;Acta2&quot;,&quot;Myh11&quot;))
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% selGenes$gene)

##make a count matrix of signature genes
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 3))
sceSub$sign2[which(sceSub$sign &gt; 3)] &lt;- 3
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 3.928997</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/signature%20cluster3%20-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/signature%20cluster3%20-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>selGenes &lt;- data.frame(gene=c(&quot;Fbln5&quot;,&quot;Eln&quot;,&quot;Actg2&quot;,&quot;Acta2&quot;,&quot;Myh11&quot;,&quot;Mfap5&quot;,&quot;Gsn&quot;,&quot;Fndc1&quot;,&quot;Col1a1&quot;,&quot;Cd34&quot;))
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% selGenes$gene)

##make a count matrix of signature genes
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 2.5))
sceSub$sign2[which(sceSub$sign &gt; 2.5)] &lt;- 2.5
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 2.554444</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/signature%20cluster3%20all-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/signature%20cluster3%20all-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>selGenes &lt;- data.frame(gene=c(&quot;Mki67&quot;, &quot;Ccna2&quot;, &quot;Cdca8&quot;, &quot;Prc1&quot;, &quot;Aurkb&quot;))
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% selGenes$gene)

##make a count matrix of signature genes
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 2))
sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 2.088394</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/proiferation%20signature-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/proiferation%20signature-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gsea" class="section level3">
<h3>GSEA</h3>
<div id="calculate-marker-genes" class="section level4">
<h4>calculate marker genes</h4>
<pre class="r"><code>## calculate marker genes
DefaultAssay(object = seuratE18EYFPv2.int) &lt;- &quot;RNA&quot;
Idents(seuratE18EYFPv2.int) &lt;- seuratE18EYFPv2.int$label
levels(seuratE18EYFPv2.int)

markerGenes &lt;- FindAllMarkers(seuratE18EYFPv2.int, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) </code></pre>
<pre class="r"><code>#save table
write.table(markerGenes, 
            file= paste0(basedir, &quot;/data/markerGenes_E18_EYFPpos_integrated_fil_label.txt&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
</div>
<div id="run-gsea-on-cluster2" class="section level4">
<h4>run GSEA on cluster2</h4>
<pre class="r"><code>## load marker genes
markerGenes &lt;- read.delim(paste0(basedir, &quot;/data/markerGenes_E18_EYFPpos_integrated_fil_label.txt&quot;), header = TRUE, sep = &quot;\t&quot;)

markerProlif &lt;- dplyr::filter(markerGenes, cluster == &quot;cluster2&quot;)
markerProlif &lt;- markerProlif%&gt;%
  mutate(Gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, gene))

##GSEA 
ego &lt;- enrichGO(gene = unique(markerProlif$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
ego &lt;- setReadable(ego, OrgDb = org.Mm.eg.db)
dotplot(ego, showCategory=30)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/GSEA%20cluster2-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="run-gsea-on-cluster3" class="section level4">
<h4>run GSEA on cluster3</h4>
<pre class="r"><code>markerProlif &lt;- dplyr::filter(markerGenes, cluster == &quot;cluster3&quot;)
markerProlif &lt;- markerProlif%&gt;%
  mutate(Gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, gene))

##GSEA Gran
ego &lt;- enrichGO(gene = unique(markerProlif$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
ego &lt;- setReadable(ego, OrgDb = org.Mm.eg.db)
dotplot(ego, showCategory=30)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/GSEA%20cluster3-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="run-gsea-on-cluster1" class="section level4">
<h4>run GSEA on cluster1</h4>
<pre class="r"><code>markerProlif &lt;- dplyr::filter(markerGenes, cluster == &quot;cluster1&quot;)
markerProlif &lt;- markerProlif%&gt;%
  mutate(Gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, gene))

##GSEA Gran
ego &lt;- enrichGO(gene = unique(markerProlif$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
ego &lt;- setReadable(ego, OrgDb = org.Mm.eg.db)
dotplot(ego, showCategory=30)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/GSEA%20cluster1-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="project-signatures" class="section level3">
<h3>project signatures</h3>
<div id="stem-cell-population-maintenance" class="section level4">
<h4>stem cell population maintenance</h4>
<pre class="r"><code>ego1 &lt;- dplyr::filter(ego@result,ego@result$Description==&quot;stem cell population maintenance&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.6))
sceSub$sign2[which(sceSub$sign &gt; 0.6)] &lt;- 0.6
sceSub$sign2[which(sceSub$sign &lt; 0)] &lt;- 0
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 0.6425939</code></pre>
<pre class="r"><code>min(sceSub$sign)</code></pre>
<pre><code>[1] 0.1083498</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/stem%20cell%20population%20maintenance-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/stem%20cell%20population%20maintenance-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  selGenes &lt;- selGenes$gene
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;.*_&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
</div>
<div id="heatmap-stem-cell-population-maintenance"
class="section level4">
<h4>heatmap stem cell population maintenance</h4>
<pre class="r"><code>sig &lt;- markerGenes %&gt;% mutate(geneID = gene) %&gt;% dplyr::filter(cluster == &quot;cluster1&quot;) %&gt;%
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) %&gt;% dplyr::filter(gene %in% df$gene)

grpCnt &lt;- sig %&gt;% group_by(cluster) %&gt;% summarise(cnt=n())
ordVec &lt;- levels(seuratE18EYFPv2.int)

pOut &lt;- avgHeatmap(seurat = seuratE18EYFPv2.int, selGenes = sig,
                  colVecIdent = colLab, 
                  ordVec=ordVec,
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/heatmap%20stem%20cell%20population%20maintenance-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="positive-regulation-of-stem-cell-population-maintenance"
class="section level4">
<h4>positive regulation of stem cell population maintenance</h4>
<pre class="r"><code>ego1 &lt;- dplyr::filter(ego@result,ego@result$Description==&quot;positive regulation of stem cell population maintenance&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.8))
sceSub$sign2[which(sceSub$sign &gt; 0.8)] &lt;- 0.8
sceSub$sign2[which(sceSub$sign &lt; 0)] &lt;- 0
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 0.8026154</code></pre>
<pre class="r"><code>min(sceSub$sign)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/positive%20regulation%20of%20stem%20cell%20population%20maintenance-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/positive%20regulation%20of%20stem%20cell%20population%20maintenance-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="heatmap-positive-regulation-of-stem-cell-population-maintenance"
class="section level4">
<h4>heatmap positive regulation of stem cell population maintenance</h4>
<pre class="r"><code>sig &lt;- markerGenes %&gt;% mutate(geneID = gene) %&gt;% dplyr::filter(cluster == &quot;cluster1&quot;) %&gt;%
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) %&gt;% dplyr::filter(gene %in% df$gene)

grpCnt &lt;- sig %&gt;% group_by(cluster) %&gt;% summarise(cnt=n())
ordVec &lt;- levels(seuratE18EYFPv2.int)

pOut &lt;- avgHeatmap(seurat = seuratE18EYFPv2.int, selGenes = sig,
                  colVecIdent = colLab, 
                  ordVec=ordVec,
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/heatmap%20positive%20regulation%20of%20stem%20cell%20population%20maintenance-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="stem-cell-proliferation" class="section level4">
<h4>stem cell proliferation</h4>
<pre class="r"><code>ego1 &lt;- dplyr::filter(ego@result,ego@result$Description==&quot;stem cell proliferation&quot;)
g1 &lt;- ego1$geneID
Str &lt;-(g1)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 0.8))
sceSub$sign2[which(sceSub$sign &gt; 0.8)] &lt;- 0.8
sceSub$sign2[which(sceSub$sign &lt; 0)] &lt;- 0
##check max and min values
max(sceSub$sign)</code></pre>
<pre><code>[1] 0.7917599</code></pre>
<pre class="r"><code>min(sceSub$sign)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;) + sc +
  theme(legend.position = &quot;none&quot;, point_size = 1)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/stem%20cell%20proliferation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) + sc</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/stem%20cell%20proliferation-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-stem-cell-proliferation" class="section level4">
<h4>heatmap stem cell proliferation</h4>
<pre class="r"><code>sig &lt;- markerGenes %&gt;% mutate(geneID = gene) %&gt;% dplyr::filter(cluster == &quot;cluster1&quot;) %&gt;%
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) %&gt;% dplyr::filter(gene %in% df$gene)

grpCnt &lt;- sig %&gt;% group_by(cluster) %&gt;% summarise(cnt=n())
ordVec &lt;- levels(seuratE18EYFPv2.int)

pOut &lt;- avgHeatmap(seurat = seuratE18EYFPv2.int, selGenes = sig,
                  colVecIdent = colLab, 
                  ordVec=ordVec,
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/heatmap%20stem%20cell%20proliferation-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="slingshot" class="section level3">
<h3>slingshot</h3>
<pre class="r"><code>sce &lt;- slingshot(sce, clusterLabels = &#39;label&#39;, reducedDim = &#39;UMAP&#39;,
                 start.clus=&quot;cluster1&quot;,
                 dist.method=&quot;simple&quot;, extend = &#39;n&#39;, stretch=0)</code></pre>
<pre class="r"><code>clustDat &lt;- data.frame(clustCol=colLab) %&gt;% rownames_to_column(., &quot;cluster&quot;)
colDat &lt;- data.frame(cluster=seuratE18EYFPv2.int$label)  %&gt;% left_join(., clustDat, by=&quot;cluster&quot;) </code></pre>
<pre class="r"><code>plot(reducedDims(sce)$UMAP, col = colDat$clustCol, pch=16, asp = 1)
lines(SlingshotDataSet(sce), lwd=2, type = &#39;lineages&#39;, col = &#39;black&#39;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/slingshot-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(reducedDims(sce)$UMAP, col = colDat$clustCol, pch=16, asp = 1)
lines(SlingshotDataSet(sce), lwd=2, col=&#39;black&#39;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/slingshot-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>colors &lt;- colorRampPalette(brewer.pal(11,&#39;YlGnBu&#39;))(100)
plotcol &lt;- colors[cut(slingAvgPseudotime(SlingshotDataSet(sce)), breaks=100)]

plot(reducedDims(sce)$UMAP, col = plotcol, pch=16, asp = 1)
lines(SlingshotDataSet(sce), lwd=2, col=&#39;black&#39;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>colors &lt;- colorRampPalette(brewer.pal(11,&#39;PuOr&#39;)[-6])(100)
plotcol &lt;- colors[cut(slingAvgPseudotime(SlingshotDataSet(sce)), breaks=100)]

plot(reducedDims(sce)$UMAP, col = plotcol, pch=16, asp = 1)
lines(SlingshotDataSet(sce), lwd=2, col=&#39;black&#39;)</code></pre>
<p><img src="figure/E18_mLN_iLN_EYFPonly_Ccl19_marker.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info" class="section level3">
<h3>session info</h3>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Tue Jul 15 17:47:55 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] future_1.58.0               RColorBrewer_1.1-3          here_1.0.1                 
 [4] slingshot_2.12.0            TrajectoryUtils_1.12.0      princurve_2.1.6            
 [7] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[10] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
[13] enrichplot_1.24.4           msigdbr_24.1.0              org.Mm.eg.db_3.19.1        
[16] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[19] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[22] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[25] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[28] forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[31] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[34] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[37] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[40] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[43] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[46] pheatmap_1.0.13             ggpubr_0.6.0                ggplot2_3.5.2              
[49] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[52] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.3                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] data.table_1.17.4         ggtree_3.12.0             R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             fastDummies_1.7.5        
 [82] gridGraphics_0.5-1        lazyeval_0.2.2            yaml_2.3.10              
 [85] survival_3.8-3            scattermore_1.2           crayon_1.5.3             
 [88] RcppAnnoy_0.0.22          progressr_0.15.1          tweenr_2.0.3             
 [91] later_1.4.2               ggridges_0.5.6            codetools_0.2-20         
 [94] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
 [97] Rtsne_0.17                shape_1.4.6.1             limma_3.60.6             
[100] pkgconfig_2.0.3           TMB_1.9.17                spatstat.univar_3.1-3    
[103] mathjaxr_1.8-0            EnvStats_3.1.0            aplot_0.2.5              
[106] scatterplot3d_0.3-44      ape_5.8-1                 spatstat.sparse_3.1-0    
[109] xtable_1.8-4              car_3.1-3                 plyr_1.8.9               
[112] httr_1.4.7                rbibutils_2.3             tools_4.4.0              
[115] globals_0.18.0            beeswarm_0.4.0            broom_1.0.8              
[118] nlme_3.1-168              lambda.r_1.2.4            assertthat_0.2.1         
[121] lme4_1.1-37               digest_0.6.37             numDeriv_2016.8-1.1      
[124] Matrix_1.7-3              farver_2.1.2              tzdb_0.5.0               
[127] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.2.0        
[130] glue_1.8.0                cachem_1.1.0              polyclip_1.10-7          
[133] generics_0.1.4            Biostrings_2.72.1         mvtnorm_1.3-3            
[136] parallelly_1.45.0         mnormt_2.1.1              statmod_1.5.0            
[139] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[142] minqa_1.2.8               pbapply_1.7-2             httr2_1.1.2              
[145] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[148] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[151] shiny_1.10.0              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[154] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[157] scales_1.4.0              R.methodsS3_1.8.2         RANN_2.6.2               
[160] Cairo_1.6-2               spatstat.data_3.1-6       rstudioapi_0.17.1        
[163] cluster_2.1.8.1           mutoss_0.1-13             spatstat.utils_3.1-4     
[166] hms_1.1.3                 fitdistrplus_1.2-2        cowplot_1.1.3            
[169] colorspace_2.1-1          rlang_1.1.6               DelayedMatrixStats_1.26.0
[172] sparseMatrixStats_1.16.0  xts_0.14.1                dotCall64_1.2            
[175] shinydashboard_0.7.3      ggforce_0.4.2             laeken_0.5.3             
[178] mgcv_1.9-3                xfun_0.52                 e1071_1.7-16             
[181] TH.data_1.1-3             iterators_1.0.14          abind_1.4-8              
[184] GOSemSim_2.30.2           treeio_1.28.0             futile.options_1.0.1     
[187] bitops_1.0-9              Rdpack_2.6.4              promises_1.3.3           
[190] scatterpie_0.2.4          RSQLite_2.4.0             qvalue_2.36.0            
[193] sandwich_3.1-1            fgsea_1.30.0              DelayedArray_0.30.1      
[196] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.0           
[199] prettyunits_1.2.0         boot_1.3-31               beachmat_2.20.0          
[202] listenv_0.9.1             Rcpp_1.0.14               edgeR_4.2.2              
[205] workflowr_1.7.1           BiocSingular_1.20.0       tensor_1.5               
[208] MASS_7.3-65               progress_1.2.3            BiocParallel_1.38.0      
[211] babelgene_22.9            spatstat.random_3.4-1     R6_2.6.1                 
[214] fastmap_1.2.0             multcomp_1.4-28           fastmatch_1.1-6          
[217] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[220] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[223] vcd_1.4-13                nnet_7.3-20               gtable_0.3.6             
[226] KernSmooth_2.23-26        miniUI_0.1.2              deldir_2.0-4             
[229] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.6.0-1            
[232] spatstat.explore_3.4-3    lifecycle_1.0.4           blme_1.0-6               
[235] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[238] robustbase_0.99-4-1       spatstat.geom_3.4-1       sn_2.1.1                 
[241] ggfun_0.1.8               future.apply_1.11.3       bslib_0.9.0              
[244] pillar_1.10.2             gplots_3.2.0              pcaMethods_1.96.0        
[247] locfit_1.5-9.12           jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] future_1.58.0               RColorBrewer_1.1-3          here_1.0.1                 
 [4] slingshot_2.12.0            TrajectoryUtils_1.12.0      princurve_2.1.6            
 [7] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[10] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
[13] enrichplot_1.24.4           msigdbr_24.1.0              org.Mm.eg.db_3.19.1        
[16] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[19] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[22] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[25] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[28] forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[31] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[34] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[37] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[40] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[43] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[46] pheatmap_1.0.13             ggpubr_0.6.0                ggplot2_3.5.2              
[49] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[52] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.3                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] data.table_1.17.4         ggtree_3.12.0             R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             fastDummies_1.7.5        
 [82] gridGraphics_0.5-1        lazyeval_0.2.2            yaml_2.3.10              
 [85] survival_3.8-3            scattermore_1.2           crayon_1.5.3             
 [88] RcppAnnoy_0.0.22          progressr_0.15.1          tweenr_2.0.3             
 [91] later_1.4.2               ggridges_0.5.6            codetools_0.2-20         
 [94] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
 [97] Rtsne_0.17                shape_1.4.6.1             limma_3.60.6             
[100] pkgconfig_2.0.3           TMB_1.9.17                spatstat.univar_3.1-3    
[103] mathjaxr_1.8-0            EnvStats_3.1.0            aplot_0.2.5              
[106] scatterplot3d_0.3-44      ape_5.8-1                 spatstat.sparse_3.1-0    
[109] xtable_1.8-4              car_3.1-3                 plyr_1.8.9               
[112] httr_1.4.7                rbibutils_2.3             tools_4.4.0              
[115] globals_0.18.0            beeswarm_0.4.0            broom_1.0.8              
[118] nlme_3.1-168              lambda.r_1.2.4            assertthat_0.2.1         
[121] lme4_1.1-37               digest_0.6.37             numDeriv_2016.8-1.1      
[124] Matrix_1.7-3              farver_2.1.2              tzdb_0.5.0               
[127] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.2.0        
[130] glue_1.8.0                cachem_1.1.0              polyclip_1.10-7          
[133] generics_0.1.4            Biostrings_2.72.1         mvtnorm_1.3-3            
[136] parallelly_1.45.0         mnormt_2.1.1              statmod_1.5.0            
[139] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[142] minqa_1.2.8               pbapply_1.7-2             httr2_1.1.2              
[145] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[148] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[151] shiny_1.10.0              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[154] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[157] scales_1.4.0              R.methodsS3_1.8.2         RANN_2.6.2               
[160] Cairo_1.6-2               spatstat.data_3.1-6       rstudioapi_0.17.1        
[163] cluster_2.1.8.1           mutoss_0.1-13             spatstat.utils_3.1-4     
[166] hms_1.1.3                 fitdistrplus_1.2-2        cowplot_1.1.3            
[169] colorspace_2.1-1          rlang_1.1.6               DelayedMatrixStats_1.26.0
[172] sparseMatrixStats_1.16.0  xts_0.14.1                dotCall64_1.2            
[175] shinydashboard_0.7.3      ggforce_0.4.2             laeken_0.5.3             
[178] mgcv_1.9-3                xfun_0.52                 e1071_1.7-16             
[181] TH.data_1.1-3             iterators_1.0.14          abind_1.4-8              
[184] GOSemSim_2.30.2           treeio_1.28.0             futile.options_1.0.1     
[187] bitops_1.0-9              Rdpack_2.6.4              promises_1.3.3           
[190] scatterpie_0.2.4          RSQLite_2.4.0             qvalue_2.36.0            
[193] sandwich_3.1-1            fgsea_1.30.0              DelayedArray_0.30.1      
[196] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.0           
[199] prettyunits_1.2.0         boot_1.3-31               beachmat_2.20.0          
[202] listenv_0.9.1             Rcpp_1.0.14               edgeR_4.2.2              
[205] workflowr_1.7.1           BiocSingular_1.20.0       tensor_1.5               
[208] MASS_7.3-65               progress_1.2.3            BiocParallel_1.38.0      
[211] babelgene_22.9            spatstat.random_3.4-1     R6_2.6.1                 
[214] fastmap_1.2.0             multcomp_1.4-28           fastmatch_1.1-6          
[217] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[220] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[223] vcd_1.4-13                nnet_7.3-20               gtable_0.3.6             
[226] KernSmooth_2.23-26        miniUI_0.1.2              deldir_2.0-4             
[229] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.6.0-1            
[232] spatstat.explore_3.4-3    lifecycle_1.0.4           blme_1.0-6               
[235] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[238] robustbase_0.99-4-1       spatstat.geom_3.4-1       sn_2.1.1                 
[241] ggfun_0.1.8               future.apply_1.11.3       bslib_0.9.0              
[244] pillar_1.10.2             gplots_3.2.0              pcaMethods_1.96.0        
[247] locfit_1.5-9.12           jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
